src/components/PeoplePicker/PeoplePicker.tsx

```diff
@@ -0,0 +1,238 @@
+            },
+            [this.props.className]
+        );
+
+        if (!this.props.errorMessage) {
+            return (
+                <input
+                    type={'text'}
+                    ref={this._ref}
+                    value={this.state.value}
+                    onBlur={this._onBlur}
+                    onFocus={this._onFocus}
+                    onChange={this._onInputChange}
+                    className={peoplePickerInputClassName}
+                    placeholder={this.props.placeholder}
+                />
+            );
+        }
+
+        return (
+            <div className="people-picker-input-error-container">
+                <Tooltip
+                    content={this.props.errorMessage}
+                    className="tooltip-error"
+                    showTooltip={this.state.isFocused}
+                    directionalHint={DirectionalHint.bottomLeftEdge}>
+                    <div className="people-picker-input-error-content">
+                        <input
+                            type={'text'}
+                            ref={this._ref}
+                            value={this.state.value}
+                            onBlur={this._onBlur}
+                            onFocus={this._onFocus}
+                            onChange={this._onInputChange}
+                            className="people-picker-input-error"
+                            placeholder={this.props.placeholder}
+                        />
+                        <Icon iconName="icon-warning2" className="people-picker-error-icon"></Icon>
+                    </div>
+                </Tooltip>
+            </div>
+        );
+    }
+
+    @autobind
+    private _renderSelectedPrincipal(): JSX.Element {
+        this.props.onSelect(this.state.selectedPrincipalList);
+
+        const peoplePickerSelectedClassName = classNames(
+            'people-picker-selected',
+            {
+                'is-active': this.state.isFocused
+            },
+            [this.props.className]
+        );
+
+        if (this.props.singleSelect) {
+            return (
+                <div>
+                    {this.state.selectedPrincipalList && this.state.selectedPrincipalList.map((principal, index) => (
+                        <Principal
+                            principal={principal}
+                            isSelected={true}
+                            onDelete={this._onSuggestionDelete}
+                        />
+                    ))}
+                </div>
+            );
+        } else {
+            return (
+                <div className={peoplePickerSelectedClassName}>
+                    {this.state.selectedPrincipalList && this.state.selectedPrincipalList.map((principal, index) => (
+                        <Principal
+                            principal={principal}
+                            isSelected={true}
+                            onDelete={this._onSuggestionDelete}
```

Comment:
You could extract this to avoid duplication.

Reply:
ok. I canged it